engine/runtime/src/main/java/org/enso/interpreter/instrument/HostObjectDebugWrapper.java

```diff
@@ -51,20 +54,18 @@ public HostObjectDebugWrapper(Object hostObject) {
    * wrapper is a string from the Truffle perspective.
    * <p>
    * Serves as a workaround for https://github.com/oracle/graal/issues/5513.
+   *
    * @param object Object to potentialy wrap in {@link HostObjectDebugWrapper}.
    */
   @TruffleBoundary
-  public static Object wrapHostValues(Object object, InteropLibrary interop) {
+  public static Object wrapHostValues(Object object, InteropLibrary interop, StructsLibrary structs) {
     if (object instanceof Atom atom) {
-      Object[] fields = atom.getFields();
+      Object[] fields = structs.getFields(atom);
```

Comment:
Yeah, this is possibly a good future improvement. I don't want to add this now, because we have no benchmarks that prove this is a problem at all, and it is significant additional complexity.